backend/src/main/java/org/cryptomator/hub/api/AuditLogResource.java

```diff
@@ -79,51 +81,57 @@ public interface AuditEventDto {
 		Instant timestamp();
 
 		static AuditEventDto fromEntity(AuditEvent entity) {
-			if (entity instanceof CreateVaultEvent cve) {
-				return new CreateVaultEventDto(cve.id, cve.timestamp, CreateVaultEvent.TYPE, cve.userId, cve.vaultId, cve.vaultName, cve.vaultDescription);
+			if (entity instanceof AddVaultMembershipEvent avme) {
+				return new AddVaultMembershipEventDto(avme.id, avme.timestamp, AddVaultMembershipEvent.TYPE, avme.addedBy, avme.vaultId, avme.authorityId);
+			} else if (entity instanceof CreateVaultEvent cve) {
+				return new CreateVaultEventDto(cve.id, cve.timestamp, CreateVaultEvent.TYPE, cve.createdBy, cve.vaultId, cve.vaultName, cve.vaultDescription);
 			} else if (entity instanceof GrantVaultAccessEvent gvae) {
-				return new GrantVaultAccessEventDto(gvae.id, gvae.timestamp, GrantVaultAccessEvent.TYPE, gvae.userId, gvae.vaultId, gvae.authorityId);
+				return new GrantVaultAccessEventDto(gvae.id, gvae.timestamp, GrantVaultAccessEvent.TYPE, gvae.grantedBy, gvae.vaultId, gvae.authorityId);
 			} else if (entity instanceof RegisterDeviceEvent rde) {
-				return new RegisterDeviceEventDto(rde.id, rde.timestamp, RegisterDeviceEvent.TYPE, rde.userId, rde.deviceId, rde.deviceName, rde.deviceType);
+				return new RegisterDeviceEventDto(rde.id, rde.timestamp, RegisterDeviceEvent.TYPE, rde.registeredBy, rde.deviceId, rde.deviceName, rde.deviceType);
 			} else if (entity instanceof RemoveDeviceEvent rde) {
-				return new RemoveDeviceEventDto(rde.id, rde.timestamp, RemoveDeviceEvent.TYPE, rde.userId, rde.deviceId);
+				return new RemoveDeviceEventDto(rde.id, rde.timestamp, RemoveDeviceEvent.TYPE, rde.removedBy, rde.deviceId);
+			} else if (entity instanceof RemoveVaultMembershipEvent rvme) {
+				return new RemoveVaultMembershipEventDto(rvme.id, rvme.timestamp, RemoveVaultMembershipEvent.TYPE, rvme.removedBy, rvme.vaultId, rvme.authorityId);
 			} else if (entity instanceof UnlockVaultEvent uve) {
-				return new UnlockVaultEventDto(uve.id, uve.timestamp, UnlockVaultEvent.TYPE, uve.userId, uve.vaultId, uve.deviceId, uve.result);
+				return new UnlockVaultEventDto(uve.id, uve.timestamp, UnlockVaultEvent.TYPE, uve.unlockedBy, uve.vaultId, uve.deviceId, uve.result);
 			} else if (entity instanceof UpdateVaultEvent uve) {
-				return new UpdateVaultEventDto(uve.id, uve.timestamp, UpdateVaultEvent.TYPE, uve.userId, uve.vaultId, uve.vaultName, uve.vaultDescription, uve.vaultArchived);
-			} else if (entity instanceof UpdateVaultMembershipEvent uvme) {
-				return new UpdateVaultMembershipEventDto(uvme.id, uvme.timestamp, UpdateVaultMembershipEvent.TYPE, uvme.userId, uvme.vaultId, uvme.authorityId, uvme.operation);
+				return new UpdateVaultEventDto(uve.id, uve.timestamp, UpdateVaultEvent.TYPE, uve.updatedBy, uve.vaultId, uve.vaultName, uve.vaultDescription, uve.vaultArchived);
 			} else {
 				throw new UnsupportedOperationException("conversion not implemented for event type " + entity.getClass());
 			}
```

Comment:
a `switch` might look better 😉

Reply:
Added TODO in 0df8420076f892e01e91ddb372e795ac40ee0c63 because I didn't want to introduce a new compilation flag in this PR.